pkgs/applications/terminal-emulators/rio/default.nix

```diff
@@ -0,0 +1,81 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+
+, autoPatchelfHook
+, cmake
+, pkg-config
+, python3
+
+, fontconfig
+, libGL
+, vulkan-loader
+
+, withX11 ? true
+, libX11
+, libXcursor
+, libXi
+, libXrandr
+, libxcb
+
+, withWayland ? false
+, wayland
+}:
+let
+  rlinkLibs = [
+    fontconfig
+    libGL
+    vulkan-loader
+  ] ++ lib.optional withX11 [
+    libX11
+    libXcursor
+    libXi
+    libXrandr
+    libxcb
+  ] ++ lib.optional withWayland [
+    wayland
+  ];
+in
+rustPlatform.buildRustPackage rec {
+  pname = "rio";
+  version = "0.0.6.1-unstable-2023-06-13";
+
+  src = fetchFromGitHub {
+    owner = "raphamorim";
+    repo = "rio";
+    rev = "fe44ac3a06cffa56c220688821c8e8eb1cad69b7";
+    hash = "sha256-A8wDv4j48fCdL6Rba2LP6FGIRf7gCa9eK7j7QfUHTsk=";
+  };
+
+  cargoHash = "sha256-laqVqe+4xtz28+Hwh0CVoRyG1Uxxx4Sz67Yy1FOJQcI=";
+
+  nativeBuildInputs = [
+    autoPatchelfHook
```

Comment:
I am curious why this is necessary. Does the source code includes some binaries?

Reply:
It doesn't find the libraries otherwise.

Reply:
I can understand this, but I am asking why?

Reply:
To explain: generally `autoPatchelfHook` is used for binaries that need to be patched, but this program looks like it is compiled from source. So I am trying to understand what is being patched here.